src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -1059,11 +1061,23 @@ public Message request(String subject, byte[] body, Duration timeout) throws Int
         String responseToken = getResponseToken(responseInbox);
         CompletableFuture<Message> future = new CompletableFuture<>();
 
-        responses.put(responseToken, future);
+        if (!oldStyle) {

```

Comment:
nitpick, could simplify with if / else and move `statistics.incrementOutstandingRequests();` earlier.